proof/refine/ARM/KHeap_R.thy

```diff
@@ -1818,16 +1856,44 @@ lemma valid_global_refs_lift':
       apply (wp ctes hoare_vcg_const_Ball_lift arch idle irqn maxObj)+
   done
 
+lemma valid_pde_mappings'_lift':
+  assumes pds: "\<And>(pd::ARM_H.pde) p P. f \<lbrace>\<lambda>s. P (ko_at' pd p s) \<rbrace>"
+  shows "\<lbrace>\<lambda>s. valid_pde_mappings' s\<rbrace> f \<lbrace>\<lambda>rv s. valid_pde_mappings' s\<rbrace>"
+  unfolding valid_pde_mappings'_def valid_pde_mapping'_def
+  by (wpsimp wp: hoare_vcg_all_lift hoare_vcg_imp_lift' pds)
+
+(* FIXME arch-split: no interface lemma prospects due to valid_pde_mappings', meaning interface
+   lemmas will be limited to preserving valid_arch_state' in most cases *)
 lemma valid_arch_state_lift':
   assumes typs: "\<And>T p P. \<lbrace>\<lambda>s. P (typ_at' T p s)\<rbrace> f \<lbrace>\<lambda>_ s. P (typ_at' T p s)\<rbrace>"
+  assumes pds: "\<And>(pd::ARM_H.pde) p P. f \<lbrace>\<lambda>s. P (ko_at' pd p s) \<rbrace>"
```

Comment:
We should have a look at whether it would make sense to designate a vspace object level on all architectures and if that would be sufficient interface for valid_arch_state. They all do have some object that happens to be at the top, for RISCV64 it would come down to all PTEs, but that might still be useful.

Anyway, for later. Happy with leaving the FIXME for this pass.